Core/Transaction.cs

```diff
@@ -3,16 +3,17 @@
 
 namespace Core
 {
-    public struct Transaction : IComparable<Transaction>
+    public class Transaction : IComparable<Transaction>
     {
         [JsonConstructor]
-        public Transaction(string cardNumber, Date date, Money amount, string description, string category)
+        public Transaction(string cardNumber, Date date, Money amount, string description, string category, int? hash = null)
```

Comment:
Codacy found an issue: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6066087)